gorm/collection_operators.go

```diff
@@ -34,7 +34,9 @@ func ApplyCollectionOperators(ctx context.Context, db *gorm.DB, obj interface{},
 		return nil, err
 	}
 
-	db = ApplyPagination(ctx, db, p)
+	if !IsPaginationEmpty(p) {
```

Comment:
Does it fix the issue? If offset is specified and limit is not, then the query will still return 1000 records. I think we agreed to entirely disable default limit.